tallypush/tally.go

```diff
@@ -93,14 +95,47 @@ type histogram struct {
 
 	// lasts keep the last value pushed to tally per histogram bucket.  This
 	// defaults to zero.
-	lasts map[int64]int64
+	lasts       []int64
+	bucketValue []int64
 }
 
+// Set is log(n) because it performs binary search to find the index that bucket belongs to
```

Comment:
Set is not log(n). The binary search is, but `ensureBucket` on L118 will incur a linear copy if you are adding a zero in the middle of the slice.  So this is O(n). We should probably leave this comment out

Reply:
Let me improve the comment. I put log(n) because my assumption was that insertions won't happen often. Buckets have a static size, so once you populate it the subsequent calls are going to be log(n).